### PR TITLE
Per validator progress

### PR DESCRIPTION
**Pull request by cskiraly**
Measure validation progress per validator, not just per node

### CODE REVIEW

File: DAS/simulator.py [hunk 1]
```diff
@@ -242,9 +242,9 @@ def run(self):
                 self.validators[i].updateStats()
             trafficStatsVector.append(trafficStats)
 
-            missingSamples, sampleProgress, nodeProgress, validatorProgress = self.glob.getProgress(self.validators)
-            self.logger.debug("step %d, arrived %0.02f %%, ready %0.02f %%, validated %0.02f %%"
-                              % (steps, sampleProgress*100, nodeProgress*100, validatorProgress*100), extra=self.format)
+            missingSamples, sampleProgress, nodeProgress, validatorAllProgress, validatorProgress = self.glob.getProgress(self.validators)
+            self.logger.info("step %d, arrived %0.02f %%, ready %0.02f %%, validatedall %0.02f %%, , validated %0.02f %%"
```
Comment (leobago) [inline DAS/simulator.py:246]: Do you want to keep this as info? Or was it just for debugging?

File: DAS/simulator.py [hunk 2]
```diff
@@ -63,15 +63,15 @@ def initValidators(self):
...
-                if i < int(heavyVal/self.shape.vpn2):  # First start with the heavy nodes
-                    start =   i  *self.shape.chi*self.shape.vpn2
-                    end   = (i+1)*self.shape.chi*self.shape.vpn2
-                else:               # Then the solo stakers
-                    j = i - int(heavyVal/self.shape.vpn2)
-                    start = offset+(  j  *self.shape.chi)
-                    end   = offset+((j+1)*self.shape.chi)
-                r = set(rows[start:end])
-                c = set(columns[start:end])
+                if i < int(lightVal/self.shape.vpn1):  # First start with the light nodes
```
Comment (leobago) [inline DAS/simulator.py:66]: Is there really a need to swap light and heavy nodes?

**Review by leobago — COMMENTED** [review]
It seems to be working but I would like to do a bit more testing. Please check the few comments I left in the code.

File: DAS/simulator.py (hunk 1 above)
Comment (cskiraly) [reply DAS/simulator.py:246]: it should be debug, good catch. I thought I've already changed it, but seemingly not :-)

File: DAS/simulator.py (hunk 2 above)
Comment (cskiraly) [reply DAS/simulator.py:66]: The of who is class1 and who is class2 is currently duplicated in two files. These should be aligned, and swapping here is one way of doing that. My testing was all with local randomness, so I've seen this only when tried to refactor the changes into PRs

File: DAS/simulator.py [hunk 3]
```diff
@@ -63,15 +63,15 @@ def initValidators(self):
...
+                    start =   i  *self.shape.chi*self.shape.vpn1
+                    end   = (i+1)*self.shape.chi*self.shape.vpn1
+                else:
+                    j = i - int(lightVal/self.shape.vpn1)
+                    start = offset+(  j  *self.shape.chi*self.shape.vpn2)
+                    end   = offset+((j+1)*self.shape.chi*self.shape.vpn2)
```
[the remark this answers is not in the document]
Comment (cskiraly) [reply DAS/simulator.py:72]: one part is the *self.shape.vpn1, another is the swap to align with class1 and class2 in validator.py.

**Pull request merged**